### PR TITLE
Upgrade dovecot to 2.3.20 to address CVEs

### PR DESCRIPTION
**Pull request by arc9693**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Upgrade dovecot to 2.3.20 to address the following:
Component: dovecot (4 CVE(s))
- CVE-2021-33515
- CVE-2021-29157
- CVE-2022-30550
- CVE-2020-28200

###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  

- https://nvd.nist.gov/vuln/detail/CVE-2021-33515
- https://nvd.nist.gov/vuln/detail/CVE-2021-29157
- https://nvd.nist.gov/vuln/detail/CVE-2022-30550
- https://nvd.nist.gov/vuln/detail/CVE-2020-28200

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=415264&view=results

### CODE REVIEW

File: SPECS-EXTENDED/dovecot/dovecot.spec [hunk 1]
```diff
@@ -164,8 +163,8 @@ autoreconf -I . -fiv #required for aarch64 support
     --with-ssl=openssl           \
     --with-ssldir=%{ssldir}      \
     --with-solr                  \
-    --with-systemdsystemunitdir=%{_unitdir}  \
-    --with-docs
+    --with-docs                  \
+    systemdsystemunitdir=%{_unitdir}
```
Comment (mfrw) [inline SPECS-EXTENDED/dovecot/dovecot.spec:167]: Are we missing a `--`:
```suggestion
    --with-systemdsystemunitdir=%{_unitdir}
```

Comment (arc9693) [reply SPECS-EXTENDED/dovecot/dovecot.spec:167]: No. It fails to compile with that a tag like structure now.

**Comment by arc9693** [comment]
Build with updates: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=415887&view=results

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**